app/javascript/packages/document-capture/components/documents-and-selfie-step.tsx

```diff
@@ -106,10 +51,13 @@ export default function DocumentsAndSelfieStep({
 }: FormStepComponentProps<DocumentsAndSelfieStepValue>) {
   const { t } = useI18n();
   const { isMobile } = useContext(DeviceContext);
-  const { isLastStep } = useContext(FormStepsContext);
+  const { isLastStep, changeStepCanComplete } = useContext(FormStepsContext);
   const { flowPath } = useContext(UploadContext);
   const { isSelfieCaptureEnabled } = useContext(SelfieCaptureContext);
-
+  const docAuthSeparatePagesEnabled = getDocAuthSeparatePagesEnabled();
+  if (isSelfieCaptureEnabled && docAuthSeparatePagesEnabled) {
+    changeStepCanComplete(false);
```

Comment:
This is a side effect in the render path, it should be in a lifecycle hook.

See: https://react.dev/learn/keeping-components-pure

Reply:
@aduth moved to a hook, LMK if this is the way you want it done